### PR TITLE
[CLI]Support remote data for pfazure CLI

### PR DESCRIPTION
**Pull request by D-W-**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

This PR fixed a bug to support remote data in pfazure CLI

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_sdk/schemas/_run.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ class RunSchema(YamlFileSchema):
     )
     connections = fields.Dict(keys=fields.Str(), values=fields.Dict(keys=fields.Str()))
     # inputs field
-    data = LocalPathField()
+    data = fields.Str()
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/schemas/_run.py:54]: We should have doc on the data we supportred in reference doc

**Review by wangchao1230 — DISMISSED** [review]

File: src/promptflow/promptflow/_sdk/schemas/_run.py (hunk 1 above)
Comment (D-W-) [reply src/promptflow/promptflow/_sdk/schemas/_run.py]: Updated in public func & class.

File: src/promptflow/promptflow/_sdk/entities/_run.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ class Run(YAMLTranslatableMixin):
     :type flow: Path
     :param name: Name of the run.
     :type name: Optional[str]
-    :param data: Input data for the run.
+    :param data: Input data for the run. Local path or remote uri(starts with azureml: or public URL) are supported.
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/entities/_run.py:54]: This is shared by local and azure. Local does not support azureml?

File: src/promptflow/promptflow/_sdk/schemas/_run.py [hunk 3]
```diff
@@ -29,6 +29,26 @@ class ResourcesSchema(metaclass=PatchedSchemaMeta):
     idle_time_before_shutdown_minutes = fields.Int()
 
 
+class RemotePathStr(fields.Str):
+    default_error_messages = {
+        "invalid_path": "Invalid remote path. "
+        "Currently only azureml://xxx or public URL(e.g. http://xxx) are supported.",
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/schemas/_run.py:35]: let's use https here

**Review by wangchao1230 — DISMISSED** [review]

File: src/promptflow/promptflow/_sdk/entities/_run.py (hunk 2 above)
Comment (D-W-) [reply src/promptflow/promptflow/_sdk/entities/_run.py]: Updated.

File: src/promptflow/promptflow/azure/_pf_client.py [hunk 4]
```diff
@@ -153,7 +153,17 @@ def run(
 
         .. note:: at least one of data or run must be provided.
 
-        .. admonition:: column_mapping
+        .. admonition::
+            Data can be local file or remote path.
+            - Example:
+                - `data = "path/to/local/file"`
```
Comment (crazygao) [inline src/promptflow/promptflow/azure/_pf_client.py:159]: shall we add tests for all these type of data?

Comment (D-W-) [reply src/promptflow/promptflow/azure/_pf_client.py:159]: yes, currently there's existing test for them: e.g. `test_submit_run_from_yaml` in `test_flow_run.py`

**Review by crazygao — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**